Source/WebKit/Platform/cocoa/WebPrivacyHelpers.mm

```diff
@@ -285,44 +219,15 @@ void requestLinkDecorationFilteringData(LinkFilteringRulesCallback&& callback)
     }];
 }
 
-StorageAccessPromptQuirkController& StorageAccessPromptQuirkController::shared()
-{
-    static MainThreadNeverDestroyed<StorageAccessPromptQuirkController> sharedInstance;
-    return sharedInstance.get();
-}
-
-Ref<StorageAccessPromptQuirkObserver> StorageAccessPromptQuirkController::observeUpdates(Function<void()>&& completionHandler)
+WPResourceType StorageAccessPromptQuirkController::resourceType() const
 {
-    ASSERT(RunLoop::isMain());
-    if (!m_notificationListener) {
-        m_notificationListener = adoptNS([WKWebPrivacyNotificationListener new]);
-        [m_notificationListener listenForStorageAccessPromptQuirkChanges:^{
-            updateQuirks([this] {
-                m_observers.forEach([](auto& observer) {
-                    observer.invokeCallback();
-                });
-            });
-        }];
-    }
-    Ref observer = StorageAccessPromptQuirkObserver::create(WTFMove(completionHandler));
-    m_observers.add(observer.get());
-    return observer;
+    return static_cast<WPResourceType>(WPResourceTypeStorageAccessPromptQuirksData);
 }
 
-void StorageAccessPromptQuirkController::setCachedQuirks(Vector<WebCore::OrganizationStorageAccessPromptQuirk>&& quirks)
+void StorageAccessPromptQuirkController::didUpdateCachedListData()
 {
-    m_cachedQuirks = WTFMove(quirks);
-    m_cachedQuirks.shrinkToFit();
-    RELEASE_LOG(ResourceLoadStatistics, "StorageAccessPromptQuirkController::setCachedQuirks: Loaded %lu storage access prompt(s) quirks from WebPrivacy.", m_cachedQuirks.size());
-}
-
-void StorageAccessPromptQuirkController::setCachedQuirksForTesting(Vector<WebCore::OrganizationStorageAccessPromptQuirk>&& quirks)
-{
-    m_wasInitialized = true;
-    setCachedQuirks(WTFMove(quirks));
-    m_observers.forEach([](auto& observer) {
-        observer.invokeCallback();
-    });
+    m_cachedListData.shrinkToFit();
+    RELEASE_LOG(ResourceLoadStatistics, "StorageAccessPromptQuirkController::setCachedQuirks: Loaded %lu storage access prompt(s) quirks from WebPrivacy.", m_cachedListData.size());
```

Comment:
> `"StorageAccessPromptQuirkController::setCachedQuirks:`

Do you still want the logs to say that?

Reply:
Oh, great catch — I should change this as well!